src/Server/Services/Jobs/JobStore.cs

```diff
@@ -0,0 +1,297 @@
+                 .ConfigureAwait(false);
+
+            operationResponse.Response.EnsureSuccessStatusCode();
+        }
+
+        private InferenceJob CreateInferenceJob(Job job, string jobName, IList<InstanceStorageInfo> instances)
+        {
+            Guard.Against.Null(job, nameof(job));
+            Guard.Against.Null(jobName, nameof(jobName));
+            Guard.Against.Null(instances, nameof(instances));
+
+            var targetStoragePath = GenerateJobPayloadsDirectory(job.JobId);
+            return new InferenceJob(targetStoragePath, job) { Instances = instances };
+        }
+
+        private string GenerateJobPayloadsDirectory(string jobId)
+        {
+            Guard.Against.NullOrWhiteSpace(jobId, nameof(jobId));
+
+            var tryCount = 0;
+            var path = string.Empty;
+            do
+            {
+                path = _fileSystem.Path.Combine(_configuration.Value.Storage.Temporary, "jobs", $"{jobId}-{DateTime.UtcNow.Millisecond}");
+                try
+                {
+                    _fileSystem.Directory.CreateDirectory(path);
+                    break;
+                }
+                catch (Exception ex)
+                {
+                    if (++tryCount > 3)
+                    {
+                        throw;
+                    }
+                    _logger.Log(LogLevel.Warning, ex, $"Failed to create job payload directory {path}");
+                }
+            } while (true);
+
+            _logger.Log(LogLevel.Information, $"Job payloads directory set to {path}");
+            return path;
+        }
+
+        private void MakeACopyOfPayload(InferenceJob request)
+        {
+            _logger.Log(LogLevel.Information, $"Copying {request.Instances.Count} instances to {request.JobPayloadsStoragePath}.");
+            var files = new Stack<InstanceStorageInfo>(request.Instances);
+            var retrySleepMs = 1000;
+            var retryCount = 0;
+
+            while (files.Count > 0)
+            {
+                try
+                {
+                    var file = files.Peek();
+                    var destPath = _fileSystem.Path.Combine(request.JobPayloadsStoragePath, $"{file.SopInstanceUid}.dcm");
+                    _fileSystem.File.Copy(file.InstanceStorageFullPath, destPath, true);
+                    _logger.Log(LogLevel.Debug, $"Instance {file.SopInstanceUid} moved to {destPath}");
+                    files.Pop();
+                }
+                catch (IOException ex) when ((ex.HResult & 0xFFFF) == 0x27 || (ex.HResult & 0xFFFF) == 0x70)
+                {
+                    if (++retryCount > 3)
+                    {
+                        _logger.Log(LogLevel.Warning, ex, $"Error copying file to {request.JobPayloadsStoragePath}; destination may be out of disk space.  Exceeded maximum retries.");
+                        break;
```

Comment:
Once this specific exception is handled and broken out, partial data is copied, but the caller will not know this fact. This means in an AI case, partial data maybe uploaded and inference run, with bad result. This is a major safety concern.
Exception should be thrown or a return status indicating partial copy, and let the caller to handle the scenario.

Reply:
Good catch!